br/pkg/lightning/backend/tidb/tidb.go

```diff
@@ -505,8 +531,12 @@ func (be *tidbBackend) execStmts(ctx context.Context, stmtTasks []stmtTask, batc
 				if common.IsRetryableError(err) && i != writeRowsMaxRetryTimes-1 {
 					continue
 				}
-				// TODO: count, record and skip the error.
-				// Just return if any error occurs for now.
+				firstRow := stmtTask.rows[0]
+				err = be.errorMgr.RecordTypeError(ctx, log.L(), tableName, firstRow.path, firstRow.offset, firstRow.insertStmt, err)
```

Comment:
Why do we only record the first statement here?

Reply:
there can only be 1 statement, since it is not batch mode (maybe we can get rid of `batch` parameter 🤔)